### PR TITLE
Fix regression in ObjectId.compareTo

### PR DESCRIPTION
**Pull request by rozza**
Now checks the byte array unsigned int values
JAVA-1785

[Review on Reviewable](https://reviewable.io/reviews/rozza/mongo-java-driver/90)

### CODE REVIEW

File: bson/src/main/org/bson/types/ObjectId.java [hunk 1]
```diff
@@ -405,22 +405,14 @@ public int compareTo(final ObjectId other) {
...
-        }
-
-        x = processIdentifier - other.processIdentifier;
-        if (x != 0) {
-            return x;
+        byte[] byteArray = toByteArray();
+        byte[] otherByteArray = other.toByteArray();
+        for (int i = 0; i < 12; i++) {
+            if (byteArray[i] != otherByteArray[i]) {
+                return Integer.compare(byteArray[i] & 0xff, otherByteArray[i] & 0xff);
```
Comment (jyemin) [inline bson/src/main/org/bson/types/ObjectId.java]: There's a Integer#compareUnsigned method

Comment (rozza) [inline bson/src/main/org/bson/types/ObjectId.java]: That was introduced in Java 8

Comment (jyemin) [inline bson/src/main/org/bson/types/ObjectId.java]: Oh, rats

**Comment by jyemin** [comment]
lgtm

**Comment by rozza** [comment]
rebased and merged

**Pull request closed without merging**